lib/matplotlib/lines.py

```diff
@@ -898,6 +925,8 @@ def set_color(self, color):
 
         ACCEPTS: any matplotlib color
         """
+        if color != self._color:
```

Comment:
As for the logic to check for change in values... I could imagine a magical decorator that would examine the name of the decorated function, figure out the property it is setting, and then automatically add the logic for comparing the original value with the new value and setting the property. Although that would get a bit tricky for any setters that obtains values from the rcParams, or any other sort of normalization.

Also might be tricky to apply everywhere if the setters are used as the way to initialize the attributes (so you can't access it prior to setting it).